resilience4j-spring/src/main/java/io/github/resilience4j/circuitbreaker/configure/CircuitBreakerAspect.java

```diff
@@ -152,6 +155,19 @@ private Object handleJoinPointCompletableFuture(ProceedingJoinPoint proceedingJo
 		});
 	}
 
+	/**
+	 * handle the Future return types AOP based into configured circuit-breaker
+	 */
+	private Object handleJoinPointFuture(ProceedingJoinPoint proceedingJoinPoint, io.github.resilience4j.circuitbreaker.CircuitBreaker circuitBreaker) {
+		return circuitBreaker.executeSupplier(() -> {
```

Comment:
Shouldn't `CircuitBreaker` have an `executeFuture()` method to call here?

While I'm here anyway, why does this code catch `Throwable` rather than `Exception`? I know this just copies behavior from nearby, but it seems troublesome.

Reply:
I will address other parts of code later after we understand how to correctly decorate Futures.